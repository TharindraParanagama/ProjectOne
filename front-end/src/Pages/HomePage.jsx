import React from "react";
import Carousel from "../Components/carouselHomePage";
import Content from "../Components/content";

export default function Home() {
  return (
    <React.Fragment>
      <Carousel />
      <Content />
    </React.Fragment>
  );
}
