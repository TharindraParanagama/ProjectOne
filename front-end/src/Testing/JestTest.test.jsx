import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import HomePage from "../Pages/HomePage";
import Content from "../components/content";
import Carousel from "../components/carousel-HomePage";

Enzyme.configure({ adapter: new Adapter() });

describe("Home Page", () => {
  const wrapper = shallow(<HomePage />);
  it("should contain a componenet named content", () => {
    const ContentComponent = wrapper.find("Content");
    expect(ContentComponent.length).toBe(1);
  });
});

describe("Content Component", () => {
  const footer = shallow(<Content />);
  it("content page to have a footer", () => {
    const foot = footer.find("footer");
    expect(foot.text()).toBe("George R.R. Martin");
  });
});

describe("Carousel", () => {
  const carousel = shallow(<Carousel />);
  it("check the number of image tags in the carousel of the home page", () => {
    const image = carousel.find("img");
    expect(image.length).toBe(3);
    expect(carousel.getElements()).toMatchSnapshot();
  });
});
