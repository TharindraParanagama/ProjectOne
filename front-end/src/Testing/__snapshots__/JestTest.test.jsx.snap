// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel check the number of image tags in the carousel of the home page 1`] = `
Array [
  <div
    className="carousel slide"
    data-interval="2000"
    data-ride="carousel"
    id="carouselExampleIndicators"
  >
    <ol
      className="carousel-indicators"
    >
      <li
        className="active"
        data-slide-to="0"
        data-target="#carouselExampleIndicators"
      />
      <li
        data-slide-to="1"
        data-target="#carouselExampleIndicators"
      />
      <li
        data-slide-to="2"
        data-target="#carouselExampleIndicators"
      />
    </ol>
    <div
      className="carousel-inner"
    >
      <div
        className="carousel-item active"
      >
        <img
          alt="First slide"
          className="d-block w-100"
          src="download.jpg"
        />
      </div>
      <div
        className="carousel-item"
      >
        <img
          alt="Second slide"
          className="d-block w-100"
          src="aup.jpg"
        />
      </div>
      <div
        className="carousel-item"
      >
        <img
          alt="Third slide"
          className="d-block w-100"
          src="search.jpg"
        />
      </div>
    </div>
    <a
      className="carousel-control-prev"
      data-slide="prev"
      href="#carouselExampleIndicators"
      role="button"
    >
      <span
        aria-hidden="true"
        className="carousel-control-prev-icon"
      />
      <span
        className="sr-only"
      >
        Previous
      </span>
    </a>
    <a
      className="carousel-control-next"
      data-slide="next"
      href="#carouselExampleIndicators"
      role="button"
    >
      <span
        aria-hidden="true"
        className="carousel-control-next-icon"
      />
      <span
        className="sr-only"
      >
        Next
      </span>
    </a>
  </div>,
]
`;
